broker/src/test/java/io/zeebe/broker/exporter/ExporterManagerTest.java

```diff
@@ -121,7 +121,7 @@ public void open(Controller controller) {
     public void export(final Record record) {
       controller.updateLastExportedRecordPosition(record.getPosition());
 
-      records.add(record);
+      records.add(record.clone());
```

Comment:
Why we need that now?

Reply:
Isn't risky not to clone? I was wondering how this wasn't a problem before actually

Reply:
Yes but normally you call to json which creates a complete copy

Reply:
Where are we calling `toJson`? I'm not sure I get it :sweat_smile: Isn't this exporter just saving all records in a list? And these records share reusable buffers? Can't that lead to some issues where the records are not what we expect?

Reply:
It worked before right :sweat_smile: :laughing:

Reply:
I can check it later

Reply:
If you remove it, add a breakpoint at the end of `shouldRestoreExporterFromState` and inspect the records, it's all the same record; still passes but probably not for the right reason :sweat_smile:

Reply:
haha nice then we should probably also fix the test :D

Reply:
ok got it because there is only the test of the record key which is equal on the record in the test. ok its fine for me